src/missing_tokens.py

```diff
@@ -28,67 +27,75 @@ def __init__(self, address: Address):
         token_contract = w3.eth.contract(
             address=Web3.toChecksumAddress(self.address),
             abi=ERC20_ABI,
-            ContractFactoryClass=Factory,
         )
-        self.symbol = token_contract.symbol()
-        self.decimals = token_contract.decimals()
+        self.symbol = token_contract.caller.symbol()
+        self.decimals = token_contract.caller.decimals()
 
     def to_str(self, version: DuneVersion):
         if version == DuneVersion.V1:
-            address_bytea = f"\\\\x{self.address[2:]}"
-            return f"{address_bytea}\t{self.symbol}\t{self.decimals}"
+            self.v1_string()
         if version == DuneVersion.V2:
-            return f"('{self.address.lower()}', '{self.symbol}', {self.decimals}),"
+            return self.v2_string()
         raise ValueError(f"Invalid DuneVersion {version}")
 
+    def v1_string(self) -> str:
+        address_bytea = f"\\\\x{self.address[2:]}"
+        return f"{address_bytea}\t{self.symbol}\t{self.decimals}"
+
+    def v2_string(self) -> str:
+        return f"('{self.address.lower()}', '{self.symbol}', {self.decimals}),"
+
+
+@dataclass
+class MissingTokenResults:
+    v1: list[Address]
+    v2: list[Address]
+
+    def is_empty(self) -> bool:
+        return self.v1 is [] and self.v2 is []
+
+    def all_tokens(self) -> set[Address]:
+        return set(self.v1 + self.v2)
 
-def fetch_missing_tokens(dune: DuneAPI, file: Optional[File]) -> list[Address]:
-    """Initiates and executes Dune query for affiliate data on given month"""
-    if file:
-        # Until we can fetch from DuneV2, this will have to be a file.
-        print(f"Loading missing tokens from: {file.filename()}")
-        with open(file.filename(), "r") as csv_file:
-            return [Address(row["token"]) for row in csv.DictReader(csv_file)]
-
-    print("Getting missing tokens from: https://dune.com/queries/236085")
-    query = DuneQuery.from_environment(
-        raw_sql=open_query("./queries/missing-tokens.sql"),
-        name=f"Missing Tokens",
-        network=Network.MAINNET,
-        parameters=[],
+
+def fetch_missing_tokens_api(dune: DuneClient) -> MissingTokenResults:
```

Comment:
In my opinion if this going to be the default method to fetch missing tokens I think the `_api` unnecessary in this case.